test/misc.jl

```diff
@@ -241,8 +241,8 @@ do_something_async_27164() = @async(begin sleep(1); x27164[] = 2; end)
 
 let t = nothing
     @sync begin
-        t = do_something_async_27164()
         @async (sleep(0.05); x27164[] = 1)
+        t = do_something_async_27164()
```

Comment:
Why use `sleep` here at all? Why not have a condition that is waited on in `do_something_async_27164` and notified after the `async`? Sure that'll cause a deadlock rather than a fail if the system has a bug, but I think as is, this could still fail if the system decides to take a long break after starting the `do_something_async_27164` task, no?

Reply:
ah, that would be much smarter